application/classes/Ushahidi/Console/Notification/SavedSearch.php

```diff
@@ -0,0 +1,180 @@
+<?php defined('SYSPATH') or die('No direct script access');
+
+/**
+ * Ushahidi Saved Search Console Command
+ * Discover and queue new posts from Saved Searches
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Console
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+use Ushahidi\Console\Command;
+
+use Ushahidi\Core\Entity\ContactRepository;
+use Ushahidi\Core\Entity\SetRepository;
+use Ushahidi\Core\Entity\PostRepository;
+use Ushahidi\Core\Entity\MessageRepository;
+
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class Ushahidi_Console_Notification_SavedSearch extends Command
+{
+	private $contactRepository;
+	private $setRepository;
+	private $postRepository;
+	private $messageRepository;
+	private $db;
+
+	public function setContactRepo(ContactRepository $repo)
+	{
+		$this->contactRepository = $repo;
+	}
+
+	public function setSetRepo(SetRepository $repo)
+	{
+		$this->setRepository = $repo;
+	}
+
+	public function setPostRepo(PostRepository $repo)
+	{
+		$this->postRepository = $repo;
+	}
+
+	public function setMessageRepo(MessageRepository $repo)
+	{
+		$this->messageRepository = $repo;
+	}
+
+	protected function configure()
+	{
+		$this
+			->setName('notification:savedsearch')
+			->setDescription('Manage notifications for Saved Searches')
+			->addArgument('action', InputArgument::OPTIONAL, 'list, queue', 'list')
+			;
+	}
+
+	protected function executeList(InputInterface $input, OutputInterface $output)
+	{
+		return [
+			[
+				'Available actions' => 'queue'
+			]
+		];
+	}
+
+	protected function executeQueue(InputInterface $input, OutputInterface $output)
+	{
+		$count = 0;
+
+		// Search data
+		$search = service('factory.data')->get('search');
+		
+		$this->db = service('kohana.db');
+
+		// Get saved searches
+		$this->setRepository->setSearchParams($search);
+
+		// @todo Might need to limit the number of saved searches retrieved at a time
+		$savedSearches = $this->setRepository->getSearchResults();
+
+		$config = Kohana::$config->load('saved_search');
+		$lastSearchTimestamp = $config->get('last_search_timestamp') ?: 0;
+		$lastSearchTimestamp = date('Y-m-d H:i:s', $lastSearchTimestamp);
+
+		// Start transaction
+		$this->db->begin();
+
+		foreach ($savedSearches as $savedSearch) {
+			// Get posts with the search filter
+			$search = service('factory.data')->get('search');
+			$search->q = $savedSearch->filter['q'];
+			$search->created_after = $lastSearchTimestamp;
```

Comment:
We talked about this a little while ago. This doesn't work since a post is likely to be created, then given a category (or some other change) later which makes it match a search.

- You might be able to use `updated_after` instead.. but I think that might end up sending dupe alerts
- I think we probably need to save the the result of a saved search, then on future runs we diff against the previous set of posts and newly added posts. Does that make sense?
We could use the posts_sets table to record this for saved searches I think. However there might be some scale issues.. :/